pkg/cloudprovider/aws/ami.go

```diff
@@ -83,13 +84,25 @@ func (p *AMIProvider) getAMIID(ctx context.Context, query string) (string, error
 	return ami, nil
 }
 
-func (p *AMIProvider) getSSMQuery(instanceType cloudprovider.InstanceType, version string) string {
+func (p *AMIProvider) getSSMQuery(ctx context.Context, constraints *v1alpha1.Constraints, instanceType cloudprovider.InstanceType, version string) string {
 	var amiSuffix string
+	var arch string
 	if !instanceType.NvidiaGPUs().IsZero() || !instanceType.AWSNeurons().IsZero() {
 		amiSuffix = "-gpu"
 	} else if instanceType.Architecture() == v1alpha5.ArchitectureArm64 {
 		amiSuffix = "-arm64"
+		arch = "arm64"
+	} else if instanceType.Architecture() == v1alpha5.ArchitectureAmd64 {
+		arch = "arm64"
 	}
+
+	if constraints.AMIFamily != nil {
```

Comment:
Switch statements don't work particularly well with pointers. I might just use an if series
```
if constraints.AMIFamily == v1alpha1.OperatingSystemBottleRocket {
  ...
} else if constraints.AMIFamily == v1alpha1.OperatingSystemEKSOptimized {
  ...
} else {
  // default behavior
}
```

Also -- we should have a value you can specify for `eks-optimized`

Reply:
I modified this - but still not sure it's correct. Feel free to correct if I'm still misunderstanding how this should be done - still learning. :)